src/Corporation/Actions.ts

```diff
@@ -251,7 +251,7 @@ function assertSmartSupplyOption(value: string): asserts value is CorpSmartSuppl
   }
 }
 
-export function SetSmartSupplyOption(warehouse: Warehouse, material: Material, useOption: string): void {
+export function SetSmartSupplyOption(warehouse: Warehouse, material: Material, useOption: CorpSmartSupplyOption): void {
   assertSmartSupplyOption(useOption);
```

Comment:
This also should either take in a string and do validation itself (worse option) or take in the correct type and expect validation to be performed before calling this function